axelor-admin/src/main/java/com/axelor/apps/base/service/app/DataBackupCreateService.java

```diff
@@ -607,10 +601,19 @@ protected String getMetaModelData(
     String propertyTypeStr = property.getType().toString();
 
     if (anonymizeData) {
-      if (metaFieldsToAnonymize != null) {
-        for (MetaField metaField : metaFieldsToAnonymize) {
-          if (property.getName().equals(metaField.getName())) {
-            return anonymizeService.anonymizeValue(value, property).toString();
+      for (DataBackupConfigAnonymizeLine dataBackupConfigAnonymizeLine :
+          dataBackupConfigAnonymizeLineList) {
+        if (metaModelName.equals(dataBackupConfigAnonymizeLine.getMetaModel().getName())) {
+          if (property.getName().equals(dataBackupConfigAnonymizeLine.getMetaField().getName())) {
```

Comment:
`dataBackupConfigAnonymizeLine.getMetaField().getName()` potential NPE if metaField is null